src/main.h

```diff
@@ -426,6 +429,7 @@ class CScriptCheck
         std::swap(nFlags, check.nFlags);
         std::swap(cacheStore, check.cacheStore);
         std::swap(error, check.error);
+        std::swap(cachedHashesMap, check.cachedHashesMap);
```

Comment:
I don't understand, you mean the internal lock of cachedHashesMap ? no, the lock is only meant to protect the internal map, which is not accessed during the swap.